src/timepicker/TimePicker.ts

```diff
@@ -166,41 +166,50 @@ export const TimePickerBase = ThemeableMixin(WidgetBase);
 export class TimePicker extends TimePickerBase<TimePickerProperties> {
 	protected options: TimeUnits[] | null;
 
-	render(): DNode {
+	private _formatUnits(units: TimeUnits): string {
+		const { step = 60 } = this.properties;
+		const { hour, minute, second } = units;
+
+		return (step >= 60 ? [ hour, minute ] : [ hour, minute, second ])
+			.map(unit => padStart(String(unit), 2, '0'))
+			.join(':');
+	}
+
+	private _getOptionLabel(value: TimeUnits) {
+		const { getOptionLabel } = this.properties;
+		const units = parseUnits(value);
+		return getOptionLabel ? getOptionLabel(units) : this._formatUnits(units);
+	}
+
+	private _onNativeBlur(event: FocusEvent) {
+		this.properties.onBlur && this.properties.onBlur((<HTMLInputElement> event.target).value);
```

Comment:
You could add this interface and use this instead of `FocusEvent` to clean things up a little.

```ts
interface FocusInputEvent extends FocusEvent {
	target: HTMLInputElement;
}
```